### PR TITLE
feat(umaverse): add fast search

### PR DESCRIPTION
**Pull request by Gamaranto**
**Motivation**

This PR adds free text search over synths and allows for using normal words when searching for synths.

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [ ]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [x]  All existing tests pass
- [ ]  Untested

Signed-off-by: Gamaranto <francesco@umaproject.org>

### CODE REVIEW

File: components/Hero.tsx [hunk 1]
```diff
@@ -1,27 +1,157 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 import styled from "@emotion/styled";
+import Image from "next/image";
+import { useQueryClient } from "react-query";
 import { useRouter } from "next/router";
+import { matchSorter } from "match-sorter";
```
Comment (daywiss) [inline components/Hero.tsx:6]: just looked into this, looks pretty good

File: components/Hero.tsx [hunk 2]
```diff
@@ -1,27 +1,157 @@
...
+    stateReducer,
+    onInputValueChange: ({ inputValue }) => {
+      if (!inputValue) {
+        setMatches([]);
+        return;
+      }
+      if (!searchMap) {
+        return;
+      }
+      setMatches(
```
Comment (daywiss) [inline components/Hero.tsx:74]: i tested searching on the site, was suprised it was pretty fast as i typed, but if this becomes a performance issue we might want to use lodash.debounce or something to defer searching till typing stops

**Review by daywiss — APPROVED** [review]

File: components/Hero.tsx (hunk 2 above)
Comment (Gamaranto) [reply components/Hero.tsx:74]: Agree! I had the debounced the function initially but couldn't see any big difference, so I've removed it to make the code simpler

File: components/Hero.tsx (hunk 1 above)
Comment (Gamaranto) [reply components/Hero.tsx:6]: Huge time saver indeed

**Pull request merged**